cilium/CFP-37315-bgp-session-with-default-gateway.md

```diff
@@ -0,0 +1,167 @@
+    peers:
+    - name: "65000"
+      peerASN: 65000
+      afi: ipv4
+      peerConfigRef:
+        name: "cilium-peer"
+```
+
+proposed config in the device (arista):
+```
+bgp listen range 10.2.0.0/24 peer-group SERVERS remote-as 65001
+router bgp 65100
+   router-id 10.1.1.1
+   timers bgp 1 3
+   neighbor SERVERS local-as 65000 no-prepend replace-as
+```
+
+### Routes from Statedb
+
+cilium agent should rely on statedb to get the default gateway to create the bgp session. we can inject statedb into neighbor reconcillation struct and when we loop through neighbors, we can populate the peer ip by looking for default gateway of 0.0.0.0/0 or ::0/0 route in routes table of statedb.
+
+
+## Impacts / Key Questions
+
+### Impact 1: Multi-Homing Scenario
+
+In multi-homed environments, we will be breaking the design of one peer per bgp session because we will create two bgp sessions with one peer defined in the config. One of the workarounds could be to define two peers with same config and a unique name.
+
+### Key Question
+
+Is there a different way to accomplish this?
+
+
+### Impact 2: Multiple BGP Instances
+
+The proposal does not take into account multiple bgp instances so auto-discovery will work in that case.
+
+### Key Question
+
+What are use-cases of having multiple bgp instances? (bgp instances is a list in config)
+
+### Impact 3: config knob for auto-discovery
+
+config knob to enable auto-discovery will be crucial because that enables the users to understand the dependencies before making the changes. We can have documentation for auto-discover and provide the pre-requisites for enabling it.
+Without the knob, the user may use auto-discovery without understanding the requirements for it.
+
+### Key Question
+
+are we okay to add a knob for auto-discovery or should we enable auto-discovery by default if peerAddress is not specified?
+
+
+### Option 1:
+
+If the user wants to create bgp sessions without specifying peerAddress, the user will remove peerAddress field, enable auto-discovery in the config and add ipv4 and/or ipv6 peers explicitly with "afi" field
+
+#### Pros
+
+Adds flexibility to create ipv4 and/or ipv6 sessions.
+auto-discovery field will help users to understand the requirements provided in the documentation before using it. (User can obviously enable it without knowing the requirements, but, we can error out with clear error messages)
+
+#### Cons
+
+For multi-homed environments, it deviates from the current "one bgp session per configured peer" design if we decide to use only one bgp peer for both sessions.
+
+### Option 2:
+
+do nothing
+
+#### Pros
+
+
+
+#### Cons
+
+User will have to manage bgp cluster files for all the k8s nodes and updates to these files can be cumbersome
+
+## Future Milestones
+
+_List things that this CFP will enable but that are out of scope for now. This can help understand the greater impact of a proposal without requiring to extend the scope of a CFP unnecessarily._
```

Comment:
We can potentially mention about the `interface` field here if we decided to not implement it in the first iteration.

https://github.com/cilium/design-cfps/pull/68/files#r1993328948

Reply:
ya. I think we should add it to future milestones. It will be better to keep it simple and iterate over it when its needed. also, adding interface field should be easy to add in the future.